numpy/polynomial/chebyshev.py

```diff
@@ -1886,6 +1887,59 @@ def chebroots(c):
     return r
 
 
+def chebinterp(func, deg, domain=None, args=()):
+    """Return the Chebyshev series instance that interpolates a given function
+    at the Chebyshev points of the first kind (mapped to the given domain).
+    The function is interpolated at the Chebyshev points of the first kind,
+    mapped onto the domain of the function being approximated.
+
+    Parameters
+    ----------
+    func : function
+        The function to be approximated. It must be a function of a single
+        variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+        that can be passed in the `args` parameter.
+    deg : int or 1-D array_like
+        Degree(s) of the fitting polynomials. If `deg` is a single
```

Comment:
Note that the result here is not the Chebyshev fit to the function, but the Chebyshev fit to the function after it is mapped to [-1, 1] (pullback), i.e., the function composition `f(mapdomain(x, [-1, 1], domain))`. That should be made clear.